### ISSUE TITLE
Can't deploy the library to non-development machine

### ISSUE THREAD

**Issue by micahmo** [sskodje/ScreenRecorderLib#25]
Hi sskodje,

I'm back again with another issue. :-) I am really loving this library and integrating it into my project. However, I am now hitting a wall when I try to deploy either the test project or my own project, which includes your library, onto a non-development system. In the case of the test app, it fails to start; in the case of my personal app, it fails to load the library. In both cases, it seems like it has a problem loading some Windows C++ dependency.

I've verified that the system has .NET 4.5 Framework, Visual C++ Redistributable 2015, and Windows 8.1 SDK installed (these are the framework versions targeted by your test project). Yet the application still fails to start up. See screenshots and stack trace at the end.

This is on a Windows Server 2012 machine (so it satisfies the Windows 8+ requirement), and I have also tested deployment on a Windows 10 machine.

I'm assuming that I am missing some sort of SDK or redist that needs to be installed on deployment machines, but haven't been able to identify what I'm missing. Once again, thanks for any help you can provide!
</br>
**.NET Framework Is Installed**
![2019-01-28 09_22_46-mdm2012deva - remote desktop connection](https://user-images.githubusercontent.com/7417301/51842240-6ee38780-22de-11e9-8939-8e562c276b9f.png)

**Visual C++ and Windows SDK Are Installed**
![2019-01-28 09_24_22-mdm2012deva - remote desktop connection](https://user-images.githubusercontent.com/7417301/51842326-9e928f80-22de-11e9-9d15-8a6552174633.png)

**Computer Info**
![2019-01-28 09_25_35-mdm2012deva - remote desktop connection](https://user-images.githubusercontent.com/7417301/51842398-c97ce380-22de-11e9-8345-506b8c56dbd2.png)

**Startup Stack Trace**
```
Application: TestApp.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NotImplementedException
   at System.Windows.Baml2006.Baml2006SchemaContext.ResolveBamlType(BamlType, Int16)
   at System.Windows.Baml2006.Baml2006SchemaContext.GetXamlType(Int16)
   at System.Windows.Baml2006.Baml2006Reader.Process_ElementStart()
   at System.Windows.Baml2006.Baml2006Reader.Process_OneBamlRecord()
   at System.Windows.Baml2006.Baml2006Reader.Process_BamlRecords()
   at System.Windows.Baml2006.Baml2006Reader.Read()
   at System.Windows.Markup.WpfXamlLoader.TransformNodes(System.Xaml.XamlReader, System.Xaml.XamlObjectWriter, Boolean, Boolean, Boolean, System.Xaml.IXamlLineInfo, System.Xaml.IXamlLineInfoConsumer, MS.Internal.Xaml.Context.XamlContextStack`1<System.Windows.Markup.WpfXamlFrame>, System.Windows.Markup.IStyleConnector)
   at System.Windows.Markup.WpfXamlLoader.Load(System.Xaml.XamlReader, System.Xaml.IXamlObjectWriterFactory, Boolean, System.Object, System.Xaml.XamlObjectWriterSettings, System.Uri)

Exception Info: System.Windows.Markup.XamlParseException
   at System.Windows.Markup.WpfXamlLoader.Load(System.Xaml.XamlReader, System.Xaml.IXamlObjectWriterFactory, Boolean, System.Object, System.Xaml.XamlObjectWriterSettings, System.Uri)
   at System.Windows.Markup.WpfXamlLoader.LoadBaml(System.Xaml.XamlReader, Boolean, System.Object, System.Xaml.Permissions.XamlAccessLevel, System.Uri)
   at System.Windows.Markup.XamlReader.LoadBaml(System.IO.Stream, System.Windows.Markup.ParserContext, System.Object, Boolean)
   at System.Windows.Application.LoadBamlStreamWithSyncInfo(System.IO.Stream, System.Windows.Markup.ParserContext)
   at System.Windows.Application.LoadComponent(System.Uri, Boolean)
   at System.Windows.Application.DoStartup()
   at System.Windows.Application.<.ctor>b__1_0(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(System.Windows.Interop.MSG ByRef)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame)
   at System.Windows.Threading.Dispatcher.PushFrame(System.Windows.Threading.DispatcherFrame)
   at System.Windows.Application.RunDispatcher(System.Object)
   at System.Windows.Application.RunInternal(System.Windows.Window)
   at System.Windows.Application.Run(System.Windows.Window)
   at System.Windows.Application.Run()
   at TestApp.App.Main()
```

**Comment by sskodje** [comment]
Weird. The only dependencies required should be .net and the c++ 2015 redistributable. I can boot up a fresh windows server 2012 image and test it.

**Comment by micahmo** [comment]
Sure, I'd appreciate that. I can deploy any dependencies that it needs, I just can't figure out what it's currently missing.

**Comment by micahmo** [comment]
It occurred to me that one big difference between my dev machine and my deployment machine is that the deployment machine is a VM. I also noticed that the deployment machine is missing some DLLs that I see are loaded by the TestApp when it runs successfully on my dev system (such as gdi32full.dll).

Do you know if this library should work on a virtual machine?

**Comment by sskodje** [comment]
I reproduced the issue and found the culprit. This library depends on Windows Media Foundation to record and encode the video, and Windows Server does not ship with media features installed by default. You must go to Server Manager ->Add Roles and Features ->Features and install Media Foundation. After doing this i confirmed the console test app was working.

**Comment by micahmo** [comment]
Ah, that's it, thank you!

I also discovered another issue which was contributing to my problem. When building the library in Debug mode, it would not run/load on a deployment system. I had to build in Release mode. That's not a problem; just something else to be aware of.

So the three things I needed were...
- Visual C++ Redist
- Media Foundation
- Release mode build

Thanks again for all your help on this!
